daffodil-core/src/main/scala/org/apache/daffodil/dsom/PropProviders.scala

```diff
@@ -107,6 +115,9 @@ final class ChainPropProvider(leafProvidersArg: Seq[LeafPropProvider], forAnnota
 
   override def toString() = Misc.getNameFromClass(this) + "(" + forAnnotation + ")"
 
+  final lazy val propertyPairsSets: Seq[Set[(String, String)]] =
```

Comment:
It's still not clear to me why this is a Seq[Set[...]]. Is this because properties can come from different locations (e.g. on the term, on a ref, on a dfdl:ref) and this captures the different properties from different locations? Why can't this be flattened?